src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -35,155 +46,608 @@ def deserialize_bom_json(file_path: Path) -> dict:
+            # Clean up existing SBOM files.
+            dep_analyzer.remove_sboms(main_ctx.component.repository.fs_path)
+
+            commands = dep_analyzer.get_cmd()
+            working_dirs: Iterable[Path] = build_tool.get_build_dirs(main_ctx.component.repository.fs_path)
+            for working_dir in working_dirs:
+                # Get the absolute path to use as the working dir in the subprocess.
+                working_dir = Path(main_ctx.component.repository.fs_path).joinpath(working_dir)
+
+                try:
+                    # Suppressing Bandit's B603 report because the repo paths are validated.
+                    analyzer_output = subprocess.run(  # nosec B603
+                        commands,
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.STDOUT,
+                        check=True,
+                        cwd=str(working_dir),
+                        timeout=defaults.getint("dependency.resolver", "timeout", fallback=1200),
+                    )
+                    with open(log_path, mode="a", encoding="utf-8") as log_file:
+                        log_file.write(analyzer_output.stdout.decode("utf-8"))
+
+                except (subprocess.CalledProcessError, subprocess.TimeoutExpired) as error:
+                    logger.error(error)
+                    with open(log_path, mode="a", encoding="utf-8") as log_file:
+                        log_file.write(error.output.decode("utf-8"))
+                except FileNotFoundError as error:
+                    logger.error(error)
+
+                # We collect the generated SBOM as a best effort, even if the build exits with errors.
+                # TODO: add improvements to help the SBOM build succeed as much as possible.
+                deps_resolved |= dep_analyzer.collect_dependencies(str(working_dir), main_ctx.component)
+
+            logger.info("Stored dependency resolver log for %s to %s.", dep_analyzer.tool_name, log_path)
+
+        # Use repo finder to find more repositories to analyze.
+        if defaults.getboolean("repofinder", "find_repos"):
+            DependencyAnalyzer._resolve_more_dependencies(deps_resolved)
+
+        return deps_resolved
+
+    @staticmethod
+    def _resolve_more_dependencies(dependencies: dict[str, DependencyInfo]) -> None:
+        """Utilise the Repo Finder to resolve the repositories of more dependencies."""
+        for item in dependencies.values():
+            if item["available"] != SCMStatus.MISSING_SCM:
+                continue
+
+            item["url"] = find_repo(item["purl"])
+            if item["url"] == "":
+                logger.debug("Failed to find url for purl: %s", item["purl"])
+            else:
+                # TODO decide how to handle possible duplicates here
+                item["available"] = SCMStatus.AVAILABLE
+                item["note"] = ""
+
+    def get_root_component(self, root_bom_path: Path) -> CDXComponent | None:
```

Comment:
`self` is not used within this method, therefore we can consider making it a `@staticmethod`.